src/core/tsi/openssl_utils.h

```diff
@@ -0,0 +1,58 @@
+//
+// Copyright 2021 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#ifndef SRC_CORE_TSI_OPENSSL_UTILS_H
+#define SRC_CORE_TSI_OPENSSL_UTILS_H
+
+#include "openssl/bio.h"
+#include "openssl/evp.h"
+#include "openssl/x509.h"
+
+namespace grpc_core {
+
+// A class for managing OpenSSL |EVP_PKEY| structures.
+class OpenSslPKey {
+ public:
+  OpenSslPKey(const char* private_key, int size);
+
+  explicit OpenSslPKey(EVP_PKEY* pkey) { private_key_ = pkey; }
+
+  ~OpenSslPKey() { EVP_PKEY_free(private_key_); }
+
+  EVP_PKEY* get_private_key() { return private_key_; }
+
+ private:
+  EVP_PKEY* private_key_ = nullptr;
```

Comment:
So if EVP_PKEY can be both private key or public key, let's just name the variable `p_key_` here.